lib/src/searchfield.dart

```diff
@@ -632,7 +639,11 @@ class _SearchFieldState<T> extends State<SearchField<T>> {
                 return;
               }
               for (final suggestion in widget.suggestions) {
-                if (suggestion.searchKey
+                if (widget.comparator != null) {
+                  if (widget.comparator!(query, suggestion.searchKey)) {
```

Comment:
@aglpy I really appreciate your efforts, And this is definitely the right approach.  But I think, instead of comparing each suggestion against input, I think we should expose a function `suggestionsBuilder` which should return a  `List<SearchFieldListItem<T>` this way we can leave the complete logic to the client. what do you think?

Reply:
Nevermind it looks great!

Reply:
Do you mean to replace optionally all the for loop with a function suggestionBuilder? Could be a great idea, so this function is something like `(String query, List<SearchFieldListItem<T>) => List<SearchFieldListItem<T>` ?

Reply:
@aglpy No i meant the function would return an iterable ` FutureOr<Iterable<SearchFieldListItem>> Function(String inputText);` so that the entire logic of returning the List<SearchFieldListItem> is left to the user.

Reply:
I got it, I think it can be a useful feature for sure.